lib/string_decoder.js

```diff
@@ -122,24 +122,33 @@ function utf8CheckIncomplete(self, buf, i) {
     return 0;
   var nb = utf8CheckByte(buf[j--]);
   if (nb >= 0) {
-    if (nb > 0)
+    if (nb > 0) {
       self.lastNeed = nb + 1 - (buf.length - j);
+      if (self.lastNeed < 0)
+        self.lastNeed = nb = 0;
```

Comment:
@indutny filed https://github.com/eslint/eslint/issues/6424 for a rule proposal.